cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorNotification.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.executor;
+
+import com.linkedin.kafka.cruisecontrol.detector.notifier.AnomalyType;
+import com.linkedin.kafka.cruisecontrol.servlet.UserTaskManager;
+
+/**
+ * A class to encapsulate notification information sent to requester associated with an execution.
+ */
+public class ExecutorNotification {
+  private ActionAgent _startedBy;       // Indicate origin of the execution, could be User or Cruise Control
+  private ActionAgent _endedBy;         // An executions can be ended by Cruise Control (due to Exception or completion) or by User
+  private long _startMs;
+  private long _endMs;
+  private UserTaskManager.UserTaskInfo _userTaskInfo;
+  private boolean _executionSucceeded;
+  private String _operation;            // Can be UserTask Endpoint or Self-healing action
+  private Throwable _exception = null;  // The Exception that ended the execution
+  private String _actionUuid = null;    // UUID associated with execution, could be for UserTask for Self-healing
```

Comment:
I see Efe has a previous comment to make the fields in Class final, I think it is a good idea, we only needs to make sure they are assigned once in constructor.

Reply:
okay I think exception for `_operation` field they can all be final.